### ISSUE TITLE
Repl exception on connecting

### ISSUE THREAD

**Issue by tailhook** [edgedb/edgedb#857]
I'm executing:
```
edgedb  -p 5656 -d tutorial --no-password
```
Where the database exists, but the user is probably doesn't. The exception is:
```
  File "/work/edb/cli/__init__.py", line 72, in cli
    admin=admin,
  File "/work/edb/repl/__init__.py", line 611, in main
    return Cli(connect_kwargs).run()
  File "/work/edb/repl/__init__.py", line 449, in run
    self.ensure_connection()
  File "/work/edb/repl/__init__.py", line 269, in ensure_connection
    print(f'Could not establish connection to {dbname}: {reason}')
UnboundLocalError: local variable 'reason' referenced before assignment
```
Probably it's somewhere `edgedb.AuthenticationError` branch, but I've not debugged too much.

**Comment by 1st1** [comment]
This seems like a trivial bug. Would appreciate you fixing it and adding a functional test--running `edgedb ...` command via `subprocess.run()` and checking that it works OK with both normal and broken queries.

**Comment by elprans** [comment]
`test_cli_repl_script` is a good starting point

**Comment by tailhook** [comment]
Seems to be fixed by 6c9c3f552
